pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go

```diff
@@ -257,6 +257,12 @@ func (a *ClusterDeploymentValidatingAdmissionHook) validateCreate(admissionSpec
 		if aws.Region == "" {
 			allErrs = append(allErrs, field.Required(awsPath.Child("region"), "must specify AWS region"))
 		}
+		for i, mp := range newObject.Spec.Compute {
+			computePath := specPath.Child("compute").Index(i)
+			if mp.Platform.AWS == nil {
```

Comment:
Why is this required? The user should be able to omit it and use the defaults.

Reply:
Looking into this now, kinda looks like default machine pool platform may not be used.

Reply:
The default machine pool is something that is applied by the installer. The order of precedence for the value of a field in the machine pool platform is as follows.
1) The value set in the machine pool platform for a specific machine pool.
2) The value set in the default machine pool platform for the cluster.
3) The default value for the field.

Reply:
My concern is that the remotemachineset controller is not using the same algorithm for determining what values it expects on the resultant machine set.